framework_tvb/tvb/adapters/visualizers/complex_imaginary_coherence.py

```diff
@@ -94,9 +95,9 @@ def get_required_memory_size(self, view_model):
         """
         Return the required memory to run this algorithm.
         """
-        input_data_h5_class, input_data_h5_path = self.load_h5_of_gid(view_model.input_data.hex)
-        with input_data_h5_class(input_data_h5_path) as input_data_h5:
-            required_memory = numpy.prod(input_data_h5.read_data_shape()) * 8
+        input_data_h5 = h5.h5_file_for_gid(view_model.input_data.hex)
+        required_memory = numpy.prod(input_data_h5.read_data_shape()) * 8
+        input_data_h5.close()
```

Comment:
Good replacement in itself, but i suggest keeping the "with" syntax

Reply:
Can I use the "with" syntax if the input_data_h5 is already a created object (and the H5 file is opened)? h5_file_for_gid returns an object.

Reply:
Yes, any Object that has Method "close" can be used with "with"

Reply:
I hope I didn't miss anything... I didn't put with where they would've been to much code under it. Also, there where a few more small files like the CorrelationCorrelationsH5 for which we could upload the HasTraits data but we also need the source gid from it, which we can't directly do so I decided to leave the usage of H5 in those cases.